CMakeLists.txt

```diff
@@ -48,6 +49,32 @@ endif()
 
 project(CUTLASS ${CUTLASS_LANGUAGES})
 
+if( CUDA_COMPILER STREQUAL "clang" )
+  if( NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang" )
+    message(FATAL_ERROR "C++ compiler must be Clang. Currently it's ${CMAKE_CXX_COMPILER_ID}" )
+  endif()
+  string(APPEND CLANG_FLAGS " --std=c++11")
```

Comment:
Hi. Thanks for this PR. It was a conscious decision that cutlass 1.3 will not include any features from C++11 and later. Do you think it is possible that we can avoid using C++11 with this PR?

Reply:
What is the minimum version CUTLASS is intended to support?

--std=c++11 is needed by clang in order to make work with the standard C++ library. A lot of it depends in standard library functions being constexpr. 
I did not intend to add anything that specifically requires c++11 features to cutlass, but clang itself does need c++11 it in order to successfully compile nontrivial CUDA code that depends on the standard library. I've just tried to build cutlass with -std=c++03 and it failed miserably with all sorts of errors in libstdc++.

CUTLASS itself does not set any specific C++ standard version during compilation, so, according to the CUDA docs NVCC will be compiling the sources with whatever is the default C++ standard picked by the host compiler.  In my case (gcc 7.3) it's c++14. So clang is effectively compiling with an older standard than nvcc.

Reply:
I think we can call this one resolved. The patch does not add any new dependencies on c++11 and the flag is needed only because standard library support for CUDA compilation in clang is very limited for standard library before c++11.